src/components/sidebar/SafeList/index.tsx

```diff
@@ -75,12 +88,83 @@ const SafeList = ({ closeDrawer }: { closeDrawer?: () => void }): ReactElement =
   const hasNoSafes = Object.keys(ownedSafes).length === 0 && Object.keys(addedSafes).length === 0
   const isWelcomePage = router.pathname === AppRoutes.welcome
 
+  const handleFetchQueued = useCallback(async () => {
+    // do not run the function if safeQueuedTxs already populated
+    if (safeTxsActions !== undefined) return
+
+    const txsActionsByChain: Record<string, Record<string, SafeActions>> = {}
+
+    // TODO: try to solve the slow iteration
+    setSafeTxsActions(txsActionsByChain)
+
+    const addActionsToTxs = (chainId: string, address: string, page: TransactionListPage, isSafeOwned: boolean) => {
+      if (page.results.length === 0) return
+
+      if (isSafeOwned) {
+        const txs = page.results.filter(isTransactionListItem)
+
+        txs.reduce((acc, tx) => {
+          if (isSignableBy(tx.transaction, wallet?.address || '')) {
+            acc[chainId] = {
+              ...(acc[chainId] || {}),
+              [address]: {
+                ...(acc[chainId]?.[address] || {}),
+                signing: Number(acc[chainId]?.[address]?.signing || 0) + 1,
+              },
+            }
+          }
+          return acc
+        }, txsActionsByChain)
+      }
+
+      txsActionsByChain[chainId] = {
+        ...(txsActionsByChain[chainId] ?? {}),
+        [address]: {
+          ...txsActionsByChain[chainId]?.[address],
+          queued: `${page.results.filter(isTransactionListItem).length}${page.next ? '+' : ''}`,
+        },
+      }
+    }
```

Comment:
I'd suggest extracting this and adding a test. It may solve the slow iteration issues by assigning values directly and not spreading.